configuration/verify-service-provider.yml

```diff
@@ -29,14 +29,12 @@ serviceEntityIds: ${SERVICE_ENTITY_IDS:-["http://verify-service-provider"]}
 # To use the Verify Service Provider with multiple services, add their entity IDs here, e.g.
 # '["service-one-entity-id", "service-two-entity-id"]'
 
-# Location where Verify Hub authentication flow begins
-hubSsoLocation: ${HUB_SSO_LOCATION:-https://compliance-tool-reference.ida.digital.cabinet-office.gov.uk/SAML2/SSO}
-
-# Location of hub metadata
-# Verify Service Provider consumes the metadata and
-# Hub Public certificates from the metadata to identify the hub
-verifyHubMetadata:
-  uri: ${HUB_METADATA_URL:-https://compliance-tool-reference.ida.digital.cabinet-office.gov.uk/SAML2/metadata/federation}
+# Verify Hub Environment. This tells the service provider where the Verify Hub
+# authentication flow begins and where to find the hub metadata the Verify Service
+# Provider consumes to identify the hub.
+# Valid values: COMPLIANCE_TOOL, INTEGRATION, PRODUCTION
+verifyHubConfiguration:
+  environment: ${HUB_ENVIRONMENT:-COMPLIANCE_TOOL}
 
```

Comment:
It's not mentioned here (or the similar file in main/resources) that you can override the hubSsoLocation or metadata uri, expectedEntityId etc. We don't really expect RPs to do this, I think, but it should probably be documented somewhere for our own benefit - the main use will be in development. I am unsure where is best to put this though (it can be seen from the tests, but this feels insufficient).

Reply:
Note as well that though you can set the environment with an environment variable, you can't override the specifics with them (this is much as before, save that you had to set, and could therefore change, the sso location and metadata uri before). Overrides can be done using a yaml file, or using Java system properties if necessary (we do not recommend or document this, but it is a dropwizard default for that to be possible).

Reply:
Yeah, I wonder if we should pull the `Advanced Configuration` section of the README out into a separate `.md` file. Then we could document all sorts of crazy configurations!